plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/data/ExecuteInsertBatchImpl.java

```diff
@@ -160,53 +160,47 @@ StringBuilder prepareQueryForStatement(
         }
         query.append(")\n\tVALUES "); //$NON-NLS-1$
         StringBuilder valuesPart = new StringBuilder(64);
-        valuesPart.append("(");
         hasKey = false;
         boolean skipBindValues = CommonUtils.toBoolean(options.get(DBSDataManipulator.OPTION_SKIP_BIND_VALUES));
-        for (int i = 0; i < attributes.length; i++) {
-            DBSAttributeBase attribute = attributes[i];
-            if (DBUtils.isPseudoAttribute(attribute) || (!useMultiRowInsert && (!allNulls && DBUtils.isNullValue(attributeValues[i])))) {
+        for (int i = 0; i < attributeValues.length; i++) {
+            DBSAttributeBase attribute = attributes[i % attributes.length];
+            if (DBUtils.isPseudoAttribute(attribute) || (!useMultiRowInsert && (!allNulls && DBUtils.isNullValue(attributeValues[i % attributes.length])))) {
                 continue;
             }
-            if (hasKey) valuesPart.append(","); //$NON-NLS-1$
-            hasKey = true;
 
-            DBDValueHandler valueHandler = handlers[i];
-            
+            if (i % attributes.length == 0) {
+                valuesPart.append("(");
+                hasKey = true;
+            }
+
+            DBDValueHandler valueHandler = handlers[i % attributes.length];
```

Comment:
Either assert that `handlers.length == attributes.length` either use `i % handlers.length`. I prefer the first one.

Reply:
Went with i % handlersf.length